### PR TITLE
Header styling

### PR DESCRIPTION
**Pull request by danalvrz**
Regarding: https://github.com/kitconcept/bfs-intranet/issues/25
- Made independent wrappers for tools, and search in Header.jsx
- Styled tools and links
- Aligned logo, page links, and search icon
- Added animations to search icons
- Added animation to search-bar when active
- Added event listener to close search-bar with `Escape` key
- Added step to auto-select input text when search-bar is active
- Fixed responsiveness considering toolbar

Recording:

https://user-images.githubusercontent.com/89805481/208200498-014f8079-9c32-43fa-b310-4671d3bebe51.mov

### CODE REVIEW

**Comment by danalvrz** [comment]
I believe the z-index of the logo could be dynamically handled also, it is just an aesthetic decision inspired by the Hematogenix example given in  https://github.com/kitconcept/volto-light-theme/issues/3 . This change can be reversed, and omitted.

File: src/theme/_header.scss [hunk 1]
```diff
@@ -90,25 +82,47 @@
...
+          transition-duration: 100ms;
         }
       }
 
       .search-bar {
         position: absolute;
+        padding-top: 2em;
         z-index: 10;
-        top: -152px;
+        top: -100%;
```
Comment (davisagli) [inline src/theme/_header.scss:115]: @danalvrz I think this is 100% of the container height, not 100% of the search bar's height. It's probably the reason there is a slight delay before the search bar becomes visible when opening it. I would try setting this to -90px to match the height.

**Comment by sneridagh** [comment]
LGTM so far! great work! We historically we are using rem's instead of em's. It's said that it's a better practice (at least the last time I checked 😅) and also Semantic UI encouraged it thorough its system.

The animation looks good to me too.

**Comment by danalvrz** [comment]
@sneridagh yes, making the change to rem was the right call! Referencing the root element is more consistent

**Review by davisagli — APPROVED** [review]

**Pull request merged**